storage/database/src/key.rs

```diff
@@ -1,39 +1,71 @@
 //! Database key abstraction; `trait Key`.
 
 //---------------------------------------------------------------------------------------------------- Import
-use std::cmp::Ordering;
+use std::{cmp::Ordering, fmt::Debug};
 
-use crate::storable::Storable;
+use crate::{storable::Storable, StorableBytes, StorableStr, StorableVec};
 
 //---------------------------------------------------------------------------------------------------- Table
 /// Database [`Table`](crate::table::Table) key metadata.
 ///
 /// Purely compile time information for database table keys.
-//
-// FIXME: this doesn't need to exist right now but
-// may be used if we implement getting values using ranges.
+///
+/// ## Comparison
+/// There are 2 differences between [`Key`] and [`Storable`]:
+/// 1. [`Key`] must be [`Sized`]
+/// 2. [`Key`] represents a [`Storable`] type that defines a comparison function
+///
+/// The database backends will use [`Key::compare`] to sort the keys
+/// within database tables.
+///
+/// [`Key::compare`] is pre-implemented as a straight byte comparison.
+///
+/// This default is overridden for numbers, which use a number comparison.
+/// For example, [`u64`] keys are sorted as `{0, 1, 2 ... 999_998, 999_999, 1_000_000}`.
+///
+/// If you would like to re-define this for number types, consider creating a
+/// wrapper type around primitives like a `struct SortU8(pub u8)` and implement
+/// [`Storable`], [`Key`], and define a custom [`Key::compare`] function.
+// FIXME:
+// implement getting values using ranges.
 // <https://github.com/Cuprate/cuprate/pull/117#discussion_r1589378104>
 pub trait Key: Storable + Sized {
-    /// The primary key type.
-    type Primary: Storable;
```

Comment:
This `Primary` type was removed. This is leftover from before, but no longer makes sense since the type implementing `Key` _is_ the primary type.

Will need changing if/when multi-map support is added.